pywr/solvers/cython_glpk_edge.pyx

```diff
@@ -0,0 +1,742 @@
+            for i in range(ncols):
+                glp_set_col_stat(self.prob, i+1, self.col_stat[global_id, i])
+
+    def reset(self):
+        # Resetting this triggers a crashing of a new basis in each scenario
+        self.is_first_solve = True
+
+    cpdef object solve(self, model):
+        t0 = time.perf_counter()
+        cdef int[:] scenario_combination
+        cdef int scenario_id
+        cdef ScenarioIndex scenario_index
+        for scenario_index in model.scenarios.combinations:
+            self._solve_scenario(model, scenario_index)
+        self.stats['total'] += time.perf_counter() - t0
+        # After solving this is always false
+        self.is_first_solve = False
+
+    @cython.boundscheck(False)
+    @cython.initializedcheck(False)
+    @cython.cdivision(True)
+    cdef object _solve_scenario(self, model, ScenarioIndex scenario_index):
+        cdef Node node
+        cdef Storage storage
+        cdef AbstractNode _node
+        cdef AbstractNodeData data
+        cdef AggregatedNode agg_node
+        cdef double min_flow
+        cdef double max_flow
+        cdef double cost
+        cdef double max_volume
+        cdef double min_volume
+        cdef double avail_volume
+        cdef double t0
+        cdef int col, col2
+        cdef int* ind
+        cdef double* val
+        cdef double lb
+        cdef double ub
+        cdef Timestep timestep
+        cdef int status, simplex_ret
+        cdef cross_domain_col
+        cdef list route
+        cdef int node_id, indptr, nedges
+        cdef double flow
+        cdef int n, m
+        cdef Py_ssize_t length
+
+        timestep = model.timestep
+        cdef list edges = self.all_edges
+        nedges = len(edges)
+        cdef list non_storages = self.non_storages
+        cdef list storages = self.storages
+        cdef list virtual_storages = self.virtual_storages
+        cdef list aggregated = self.aggregated
+
+        # update route cost
+
+        t0 = time.perf_counter()
+
+        # Initialise the cost on each edge to zero
+        cdef double[:] edge_costs = self.edge_cost_arr
+        for col in range(nedges):
+            edge_costs[col] = 0.0
+
+        # update the cost of each node in the model
+        for _node in self.all_nodes:
+            cost = _node.get_cost(scenario_index)
+            data = _node.__data
+
+            if data.is_link:
+                cost /= 2
+
+            for col in data.in_edges:
+                edge_costs[col] += cost
+            for col in data.out_edges:
+                edge_costs[col] += cost
+
+        # calculate the total cost of each route
+        for col in range(nedges):
+            set_obj_coef(self.prob, self.idx_col_edges+col, edge_costs[col])
+
+        self.stats['objective_update'] += time.perf_counter() - t0
+        t0 = time.perf_counter()
+
+        # update non-storage properties
+        for col, node in enumerate(non_storages):
```

Comment:
This `col` should be `row`. Problem in other solver too? Need an issue for this.